cfgov/v1/models/base.py

```diff
@@ -158,46 +158,49 @@ def generate_view_more_url(self, request):
     def related_posts(self, block, hostname):
         from . import AbstractFilterPage
         related = {}
-        queries = {}
         query = models.Q(('tags__name__in', self.tags.names()))
         search_types = [
             ('blog', 'posts', 'Blog', query),
             ('newsroom', 'newsroom', 'Newsroom', query),
             ('events', 'events', 'Events', query),
         ]
-        for parent_slug, search_type, search_type_name, search_query in search_types:
-            try:
-                parent = Page.objects.get(slug=parent_slug)
-                search_query &= Page.objects.child_of_q(parent)
+
+        def fetch_children_by_specific_category(block, parent_slug):
+            # This used to be a Page.objects.get, which would throw
+            # an exception if the requested parent wasn't found. As of
+            # Django 1.6, you can now do Page.objects.filter().first();
+            # the advantage here is that you can check for None right
+            # away and not have to rely on catching exceptions, which
+            # in any case didn't do anything useful other than to print
+            # an error message. Instead, we just return an empty query
+            # which has no effect on the final result.
```

Comment:
Minor thing but this comment could be a multiline docstring e.g. """ comment """